src/mixxx.cpp

```diff
@@ -693,7 +692,10 @@ void MixxxMainWindow::finalize() {
     // outside of MixxxMainWindow the parent relationship will directly destroy
     // the WMainMenuBar and this will no longer be a problem.
     qDebug() << t.elapsed(false).debugMillisWithUnit() << "deleting menubar";
-    QPointer<QWidget> pMenuBar(menuBar());
+
+    QPointer<WMainMenuBar> pMenuBar = m_pMenuBar.toWeakRef();
```

Comment:
This code is hard to understand. Please add a comment why all this is necessary. Switching between direct access to m_pMenuBar and menuBar()/setMenuBar() makes it even worse.

Reply:
I have not full understand the issue myself, so I am not able to give here reasonable comments. 
For my impression the comment above is outdated and does not longer fully apply. 

But investigation this will take quite an amount of time I am not yet willing to spend, because it is target depending window manager code. 

Can we just merge this as it is, since it fixes a crasher on MacOs?

Reply:
> Switching between direct access to m_pMenuBar and menuBar()/setMenuBar() makes it even worse.

That was the issue in the original code causing the confusion that leads to the error. 
II have removed this and only used m_pMenuBar.

I have just kept the menuBar() call it in a debug assertion to verify my assumption that the old logic remains valid.

Reply:
Then please add a link to the issue and a remark that this code fixes the issue ***somehow***. Otherwise a developer might be tempted to *simplify* this code and cause a regression.